packages/devextreme/js/__internal/ui/list/m_list.base.ts

```diff
@@ -73,7 +79,9 @@ const SELECT_ALL_ITEM_SELECTOR = '.dx-list-select-all';
 const LIST_ITEM_DATA_KEY = 'dxListItemData';
 const LIST_FEEDBACK_SHOW_TIMEOUT = 70;
 
-const groupItemsGetter = compileGetter('items');
+export type Key = string | number;
```

Comment:
Let's declare ItemKey on CollectionWidgetBase level and use it everywhere in collection.
Don't forget replace TKey = string | number to TKey = ItemKey